Reporting/M5 Document/Reflection.md

```diff
@@ -0,0 +1,29 @@
+## How did your project management work for the team?  What was the hardest thing, and what would you do the same/differently the next time you plan to complete a project like this? 
+
+- Our project management worked relatively well for the team. We maintained regular communication channels, utilized project management tools effectively, and delegated tasks efficiently. However, the hardest aspect was ensuring effective coordination between front-end and back-end development teams, especially when integrating complex features like peer-to-peer messaging and encryption. 
+
+- In future projects, we would prioritize more frequent cross-team check-ins and perhaps consider adopting a more integrated development approach from the outset to mitigate such challenges.
+
+## Do you feel that your initial requirements were sufficiently detailed for this project?  Which requirements did you miss or overlook?
+- While our initial requirements provided a solid foundation for the project, there were some areas where additional detail could have been beneficial. Specifically, we overlooked the specifics of password security measures, and multimedia messaging capabilities. 
+
+- In the future, we would ensure a more comprehensive review of security and messaging features during the requirement gathering phase to avoid oversights.
+
+## What did you miss in your initial planning for the project (beyond just the requirements)?
+- In our initial planning, we may have overlooked the potential impact of external dependencies and integration challenges. Specifically, we did not anticipate the complexities associated with integrating encryption protocols into our messaging system or the time required for thorough testing across different platforms and devices.
+
+- In the future, we would allocate more time and resources to address these aspects during the planning phase to minimize delays during development.
+
+## Would you (as a team) deal with testing differently in the future?
+
+- Looking back, we recognize that our testing approach could have been more comprehensive. While we conducted unit tests and integration tests throughout the development process, we underestimated the importance of user acceptance testing (UAT) and end-to-end testing.
```

Comment:
I agree. We essentially did no user acceptance testing or end-to-end testing. We mainly relied on ad-hoc testing and had some Jest unit tests implemented as well, but those were done more earlier on before more complicated features were made. That does make sense, indeed, as it actually becomes impossible and completely impractical to write Jest tests for extremely complicated features like the `New Chat Modal` where Jest cannot even simulate dropdowns or the dependency of a second drop down on the first drop down and complex real-time database operations.